modules/m4d-s3-to-s3/templates/batchtransfer.yaml

```diff
@@ -26,13 +26,28 @@ spec:
       dataFormat: "{{ .Values.destination.format }}"
       vaultPath: "{{ .Values.destination.credentialLocation }}"
   transformation:
-  {{- range $transformations := .Values.transformations }}
-  - name: "{{ $transformations.name }}"
-    action: "{{ $transformations.action }}"
-    columns: {{ $transformations.columns }}
-  {{- end }}
-  secretProviderURL: "http://secret-provider.secret-provider.svc.cluster.local:5555/get-secret"
+  {{ range .Values.transformations }}
+  {{ if eq .id "redact-ID" }}
+  - action: "RedactColumns"
+    description: "redacting column: {{ .args.column_name }}"
+    columns: [ {{ .args.column_name | quote }} ]
+    options:
+      redactValue: "XXXXXX"
+  {{ end }}
+  {{ if eq .id "removed-ID" }}
+  - action: "RemoveColumns"
+    description: "redacting column: {{ .args.column_name }}"
+    columns: [ "{{ .args.column_name }}" ]
+  {{ end }}
+  {{ end }}
+  {{ if .Values.secretProviderURL }}
+  secretProviderURL: {{ .Values.secretProviderURL | quote }}
+  {{ end }}
   secretProviderRole: demo
```

Comment:
We have made changed such that `credentialLocation` (e.g., `.source.credentialLocation`) is used instead to point to the full URL to obtain the credentials from. It includes the role as well.